core/corerepo/gc.go

```diff
@@ -196,7 +194,14 @@ func (gc *GC) maybeGC(ctx context.Context, offset uint64) error {
 			return err
 		}
 		log.Infof("Repo GC done. Released %s\n", humanize.Bytes(uint64(storage-newStorage)))
-		return nil
+		if newStorage > gc.StorageGC {
+			log.Warningf("post-GC: Watermark still exceeded")
+			if newStorage > gc.StorageMax {
+				err := ErrMaxStorageExceeded
+				log.Error(err)
```

Comment:
this will already get logged in https://github.com/ipfs/go-ipfs/pull/2010/files#diff-1bdfa2c57df84e2debf3831f2bcecca5R157,  no?